_posts/2020-04-08-secure-software-development-lifecycle.md

```diff
@@ -0,0 +1,137 @@
+
+As a software developer and security enthusiastic I could go on forever on this topic but more or less I will just put a list here things that you should consider:
+* Know or setup secure coding practices for your language and frameworks
+* Validate all the 3rd party libraries you take in
+* Fail securely
+* Make sure that your secrets management endures the daylight
+* Know the http security headers or atleast know how to check them
+* Understand the cryptography you are using
+* Basically memorize the whole OWASP TOP 10 and constantly consider them as a threats
+* Use XP programming
+* Have a review process
+* Find tooling that supports your work
+* Sanitize inputs
+* Sanitize outputs
+
+#### Test
+
+Security testing is one that works relatively same as every testing. You should just remember that now that we have security requirements we will also need to verify that we fulfill them. Some helpful things to do is to setup a good continuous integration environment that gives your development team constant feedback about how they are doing. If they are more on the DevOps or DevSecOps then they most likely will build it by themselves but some organization have different people for these tasks. I consider continuous integration as a testing as it is something that is outsourced for a system that gives you a feedback about your work. Some could also think that it is part of the implementation. 
+
+Things that you can automatically test that help security:
+* Performance
+* Security headers
+* TLS and its configurations
+* Unit tests (also integration test, end-to-end tests, smoke tests and all the similar)
+* Known vulnerability analysis for your dependencies
+* Fuzz testing
+* Static code analysis for your source code
+* Best practice analysis in some cases
+
+You should not forget the manual testing. For example:
+* Review what others have done
+* Exploratory testing
+* Penetration testing
+* Trying out disaster recovery practices
+
+#### Deployment
+
+The moment you publish your source code as a software it becomes legacy and you need to support it. From SSDLC perspective this means things that you need actively do and things that you need to have processes for. 
```

Comment:
"...this means things that you need actively do and things that you need to have processes for" -> "this means that you need to actively do the things that you need to have processes for"